packages/measured-signalfx-reporter/lib/reporters/SignalFxMetricsReporter.js

```diff
@@ -0,0 +1,222 @@
+        dimensions: mergedDimensions
+      };
+
+      if (Object.prototype.hasOwnProperty.call(newRequest, metricValueTypeWrapper.type)) {
+        newRequest[metricValueTypeWrapper.type].push(signalFxDataPointMetric);
+      } else {
+        newRequest[metricValueTypeWrapper.type] = [signalFxDataPointMetric];
+      }
+    });
+
+    return newRequest;
+  }
+
+  /**
+   * Maps Measured Metrics Object JSON outputs to there respective signal fx metrics using logic from
+   * com.signalfx.codahale.reporter.AggregateMetricSenderSessionWrapper in the java lib to derive naming
+   *
+   * @param {string} name The registered metric base name
+   * @param {Metric} metric The metric.
+   * @return {MetricValueTypeWrapper[]} an array of MetricValueTypeWrapper that can be used to
+   * build the sig fx data point request
+   * @protected
+   */
+  _getValuesToProcessForType(name, metric) {
+    const type = metric.getType();
+    switch (type) {
+      case MetricTypes.TIMER:
+        return this._getValuesToProcessForTimer(name, metric);
+      case MetricTypes.GAUGE:
+        return this._getValuesToProcessForGauge(name, metric);
+      case MetricTypes.COUNTER:
+        return this._getValuesToProcessForCounter(name, metric);
+      case MetricTypes.HISTOGRAM:
+        return this._getValuesToProcessForHistogram(name, metric);
+      case MetricTypes.METER:
+        this._log.warn(
+          'Meters are not implemented in the signal fx reporter as you can use a counter and reduce your DPM'
+        );
+        return [];
+      default:
+        this._log.error(`Metric Type: ${type} has not been implemented to report to signal fx`);
+        return [];
+    }
+  }
+
+  /**
+   * Maps and Filters values from a Timer to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Timer} timer The Timer
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForTimer(name, timer) {
+    let valuesToProcess = [];
+    // only grab histogram data as Meters can be accomplished with signal fx using the count from the histogram
+    valuesToProcess = valuesToProcess.concat(this._getValuesToProcessForHistogram(name, timer._histogram));
+    return valuesToProcess;
+  }
+
+  /**
+   * Maps and Filters values from a Gauge to a set of metrics to report to SigFx.
+   *
+   * @param {string} name The registry name
+   * @param {Gauge} gauge The Gauge
+   * @return {MetricValueTypeWrapper[]} Returns an array of MetricValueTypeWrapper to use to build the request
+   * @protected
+   */
+  _getValuesToProcessForGauge(name, gauge) {
+    const valuesToProcess = [];
+    valuesToProcess.push({
+      metric: `${name}`,
+      value: gauge.toJSON(),
+      type: SIGNAL_FX_GAUGE
+    });
+    return valuesToProcess;
+  }
+
+  /**
+   * Maps and Filters values from a Counter to a set of metrics to report to SigFx.
```

Comment:
"SigFx" to "SignalFx"

Also, Counters and Gauges aren't filtered are they?  Is that language from Histogram and Timer?

Reply:
- [ ] Update language.